op-proposer/proposer/config.go

```diff
@@ -82,6 +83,38 @@ type CLIConfig struct {
 	WaitNodeSync bool
 }
 
+// Check checks the CLI config.
```

Comment:
nit: While this documentation is accurate, I think it's better to be less specific the CLI config requirements. So as to avoid a desync between future code changes and the code comment.
The CLI flags should already be self-documenting on what a valid config looks like.